Prox/Prox/Utilities/LocationMonitor.swift

```diff
@@ -141,7 +141,10 @@ class LocationMonitor: NSObject {
 
 extension LocationMonitor: LocationProvider {
     func getCurrentLocation() -> CLLocation? {
-        return currentLocation ?? locationManager.location
+        if currentLocation == nil {
+            currentLocation = self.locationManager.location
```

Comment:
Yup, the rest of the code handles the nil location case, this is just resetting the currentLocation on start